### PR TITLE
Fix: Keys containing dots are not translated in application

### PR DESCRIPTION
**Pull request by qmonmert**
Fix issue https://github.com/jhipster/generator-jhipster/issues/10678 in the generator

### CODE REVIEW

File: tests/spec/language/translate.spec.tsx [hunk 1]
```diff
@@ -22,7 +22,18 @@ describe('Translate', () => {
           foofoo: 'text {{ foo}} this {{bar}} <b>test</b>',
           foodirty: 'text {{ foo }} this {{bar}} <script>execute</script><br/><hr><div>test</div> <a href="test">link</a>',
           'baz.foo': 'dirty key',
-          foozfooz: 'jhipster is <strong>awesome</strong>!'
+          foozfooz: 'jhipster is <strong>awesome</strong>!',
+          'bar1.bar2': {
```
Comment (deepu105) [inline tests/spec/language/translate.spec.tsx]: Do we have such paths? I remember we only had 'foo.bar' like stuff, which were left over from legacy code

Comment (pascalgrimaud) [reply tests/spec/language/translate.spec.tsx]: I don't think we have this kind of path, but as this new code takes care of this, I think it's good.
Here a sample of what we currently have: https://github.com/jhipster/generator-jhipster/blob/master/generators/languages/templates/src/main/webapp/i18n/en/global.json.ejs#L148-L150

**Comment by pascalgrimaud** [comment]
> The code looks fine but I would rather disallow such keys than complicate our code(also affects performance). The reason we originally did was was we had some legacy keys which were invalid. We should change those in the next major release to valid json and keep this simple

You're right, @deepu105, I think it's time to discuss about a real roadmap about v7, what we can refactor / change / break

File: tests/spec/language/translate.spec.tsx (hunk 1 above)
Comment (deepu105) [reply tests/spec/language/translate.spec.tsx]: Yes, that was my point, why add this code if we don't have this kind of path? and we should never add this kind of path. So why fix something that is not broken

**Comment by deepu105** [comment]
@pascalgrimaud let's create a ticket to track v7 roadmap items

File: tests/spec/language/translate.spec.tsx (hunk 1 above)
Comment (pascalgrimaud) [reply tests/spec/language/translate.spec.tsx]: it's not a code here, it's a test (as it's in spec file), so now we're sure this kind of thing can't broke our front anymore

**Comment by pascalgrimaud** [comment]
@deepu105 : oh maybe a 'project' under generator-jhipster repo, like you did for v6 ?

**Comment by deepu105** [comment]
I created a ticket, I think its easier to collaborate then Project. The
project really didn't help much, unless you have really fine tuned tickets

Thanks & Regards,
Deepu

**Comment by pascalgrimaud** [comment]
Ok, so what the state for this ticket ? As the code looks good, I think we can merge + release react-jhipster, so it will fix the original bug.
Then, for v7, we can improve this part. What do you think ?

File: tests/spec/language/translate.spec.tsx (hunk 1 above)
Comment (deepu105) [reply tests/spec/language/translate.spec.tsx]: I mean the test is fine but I don't see what is the bug here

**Comment by deepu105** [comment]
@pascalgrimaud I think maybe we misunderstood each other, so let me try to clarify :smiley_cat: 
Is there any existing translation that is broken? when I originally coded this, I accounted for the existing dirty keys and it was working, as you can see from the tests. So unless someone added new lazy keys which were one more level nested, or I missed such a case the original code should be enough. So looking at the ticket which is being solved, were those keys already in the app or is it something new?

Anyway regardless, ok to merge this for now and we can clean it up for v7

**Comment by pascalgrimaud** [comment]
@deepu105 : yes some translations are broken with React. See this ticket https://github.com/jhipster/generator-jhipster/issues/10678 for details -> I managed to reproduce the issue, reported by @SoniaLahi, so I confirm what she found

**Comment by deepu105** [comment]
Ok then lets merge this for now

Thanks & Regards,
Deepu

**Comment by pascalgrimaud** [comment]
thanks, I'll try to do the release tonight

**Comment by qmonmert** [comment]
@deepu105 first breaking changes **Remove dirty keys in i18n JSON files and cleanup translation logic in libs** done #10958

**Pull request merged**